tesseract_scene_graph/test/tesseract_scene_graph_srdf_unit.cpp

```diff
@@ -222,6 +222,69 @@ TEST(TesseractSceneGraphSRDFUnit, LoadSRDFUnit)  // NOLINT
   EXPECT_EQ(acm->getAllAllowedCollisions().size(), 0);
 }
 
+TEST(TesseractSceneGraphSRDFUnit, LoadSRDFAllowedCollisionMatrixUnit)  // NOLINT
+{
+  using namespace tesseract_scene_graph;
+  SceneGraph::Ptr g = getABBSceneGraph();
+
+  std::string xml_string =
+      R"(<robot name="abb_irb2400">
+           <disable_collisions link1="base_link" link2="link_1" reason="Adjacent" />
+           <disable_collisions link1="base_link" link2="link_2" reason="Never" />
+           <disable_collisions link1="base_link" link2="link_3" reason="Never" />
+         </robot>)";
+  tinyxml2::XMLDocument xml_doc;
+  EXPECT_TRUE(xml_doc.Parse(xml_string.c_str()) == tinyxml2::XML_SUCCESS);
+
+  tinyxml2::XMLElement* element = xml_doc.FirstChildElement("robot");
+  EXPECT_TRUE(element != nullptr);
+
+  AllowedCollisionMatrix acm = parseDisabledCollisions(*g, element, std::array<int, 3>({ 1, 0, 0 }));
+  EXPECT_TRUE(acm.isCollisionAllowed("base_link", "link_1"));
+  EXPECT_TRUE(acm.isCollisionAllowed("base_link", "link_2"));
+  EXPECT_TRUE(acm.isCollisionAllowed("base_link", "link_3"));
+
+  // Now test failures
+  auto failure_test = [g](const std::string& xml_string) {
```

Comment:
For these, maybe rename to is_failure, or add some documentation that make it clear that returning true is a failure.

Reply:
renamed.